### PR TITLE
Serializers deprecation

### PR DESCRIPTION
**Pull request by verult**
This PR deprecates all serializers we plan to deprecate
* Deprecated GateOpSerializer, GateOpDeserializer, DeserializingArg, and SerializingArg using the class wrapper technique. 
* CircuitSerializer and serializer classes it depends on (`CircuitOp[De]Serializer`, `Serializer`) are left in place.
* Skipping deprecation of common serializers in `common_serializers.py` for now and remove them directly after the 0.15 release as it's very unlikely for users to access them directly. Most users interact with them via global gatesets, e.g.

https://github.com/quantumlib/Cirq/blob/abfc619bfe31e93573993c76cecfddc5689dea9b/cirq-google/cirq_google/serialization/gate_sets.py#L43-L64

But if time allows, I'll try to deprecate common serializers as well.

* Added mentions of serializer deprecations in global gateset deprecation warnings.

@dstrain115

### CODE REVIEW

File: cirq-google/cirq_google/__init__.py [hunk 1]
```diff
@@ -174,7 +175,8 @@
         ),
         'FSIM_GATESET': (
             'v0.16',
-            'SerializableGateSet will no longer be supported.'
+            'SerializableGateSet and associated classes (GateOpSerializer, GateOpDeserializer,'
```
Comment (dstrain115) [inline cirq-google/cirq_google/__init__.py:178]: This message is getting pretty long.  Should we make it a constant?

File: cirq-google/cirq_google/devices/known_devices.py [hunk 2]
```diff
@@ -175,7 +175,7 @@ def create_device_proto_for_qubits(
                 gate = gs_proto.valid_gates.add()
                 gate.id = gate_id
 
-                if not isinstance(serializer, op_serializer.GateOpSerializer):
+                if not isinstance(serializer, op_serializer._GateOpSerializer):
```
Comment (dstrain115) [inline cirq-google/cirq_google/devices/known_devices.py:178]: This needs to handle both GateOpSerializer and _GateOpSerilalzer, right?

**Review by dstrain115 — APPROVED** [review]
LGTM, minus two small comments.

File: cirq-google/cirq_google/devices/known_devices.py (hunk 2 above)
Comment (verult) [reply cirq-google/cirq_google/devices/known_devices.py:178]: `_GateOpSerializer` is a superclass of `GateOpSerializer` so both are handled

File: cirq-google/cirq_google/__init__.py (hunk 1 above)
Comment (verult) [reply cirq-google/cirq_google/__init__.py:178]: Ack, I added two identical constants to avoid circular dependencies.

**Comment by verult** [comment]
Thanks for the review, merging now.

**Pull request merged**